internal/lookout/ui/src/services/ComputeResourcesService.ts

```diff
@@ -0,0 +1,111 @@
+import { load } from "js-yaml"
+
+// A simple interface that our PodSpec has.
+export interface JobYaml {
+  podSpec: {
+    containers: Container[]
+  }
+}
+
+export interface ContainerInfo {
+  command?: string[]
+  arguments?: string[]
+  cpu?: string[]
+  memory?: string[]
+  gpu?: string[]
+  "ephermal-storage"?: string[]
+}
+
+interface Container {
+  command: string[]
+  args: string[]
+  resources: {
+    limits: Limits
+  }
+}
+
+interface Limits {
+  cpu?: string
+  "nvidia.com/gpu"?: string
+  "ephemeral-storage"?: string
+  memory?: string
+}
+
+export function getContainerInfoFromYaml(jobYaml: string): ContainerInfo {
+  const yaml = load(jobYaml) as JobYaml
+  return {
+    command: getCommandFromJobYaml(yaml),
```

Comment:
This logic should be changed based on previous comment, but in general you can implement this with a single map over the containers rather than having to get each list of fields individually.